diff --git a/nix-darwin/default.nix b/nix-darwin/default.nix
--- a/nix-darwin/default.nix
+++ b/nix-darwin/default.nix
@@ -29,7 +29,7 @@ in
             if [[ -v DRY_RUN || "$hmParentArgs" == *" --dry-run"* ]]; then
               hmDryRunArgs=(env DRY_RUN=1)
             fi
-            launchctl asuser "$(id -u ${usercfg.home.username})" sudo -u ${usercfg.home.username} --set-home "''${hmDryRunArgs[@]}" ${pkgs.writeShellScript "activation-${usercfg.home.username}" ''
+            launchctl asuser "$(id -u ${usercfg.home.username})" sudo --preserve-env=PATH -u ${usercfg.home.username} --set-home "''${hmDryRunArgs[@]}" ${pkgs.writeShellScript "activation-${usercfg.home.username}" ''
               ${lib.optionalString (
                 cfg.backupFileExtension != null
               ) "export HOME_MANAGER_BACKUP_EXT=${lib.escapeShellArg cfg.backupFileExtension}"}